pkg/controller/direct/datacatalog/tagtemplate_controller.go

```diff
@@ -0,0 +1,305 @@
+		// We also need to ensure that the fields are reconciled.
+		// TODO: Implement reconciliation logic for TagTemplateFields if they are managed inline.
+		// For now, just return the current actual state.
+		updated = a.actual
+	} else {
+		req := &pb.UpdateTagTemplateRequest{
+			TagTemplate: desired,
+			UpdateMask:  updateMask,
+		}
+		updated, err = a.gcpClient.UpdateTagTemplate(ctx, req)
+		if err != nil {
+			return fmt.Errorf("updating datacatalog tagtemplate %s: %w", a.id.String(), err)
+		}
+		log.V(2).Info("successfully updated datacatalog tagtemplate in gcp", "name", a.id)
+		// TODO: After successful update, reconcile TagTemplateFields if managed inline.
+	}
+
+	// Regardless of whether an update call was made, reconcile status.
+	// Need the latest state which might include output-only fields not present in `desired`.
+	// Re-fetch if no update was performed or if the update response might be incomplete.
+	if len(updateMask.Paths) == 0 {
+		refetched, findErr := a.gcpClient.GetTagTemplate(ctx, &pb.GetTagTemplateRequest{Name: a.id.String()})
+		if findErr != nil {
+			// Log the error but continue with the potentially stale 'actual' for status update
+			log.Error(findErr, "failed to re-fetch tagtemplate after no-op update", "name", a.id)
+		} else {
+			updated = refetched
+			a.actual = refetched // Update internal 'actual' state
+		}
+	} else {
+		// If an update was made, the 'updated' response should be sufficient.
+		a.actual = updated // Update internal 'actual' state
+	}
+
+	status := &krm.DataCatalogTagTemplateStatus{}
+	status.ObservedState = DataCatalogTagTemplateObservedState_FromProto(mapCtx, updated)
+	if mapCtx.Err() != nil {
+		return mapCtx.Err()
+	}
+	return updateOp.UpdateStatus(ctx, status, nil)
+}
+
+func (a *tagTemplateAdapter) Export(ctx context.Context) (*unstructured.Unstructured, error) {
+	if a.actual == nil {
+		return nil, fmt.Errorf("Find() not called")
+	}
+	u := &unstructured.Unstructured{}
+
+	obj := &krm.DataCatalogTagTemplate{}
+	mapCtx := &direct.MapContext{}
+	// Export the observed state, including fields.
+	obj.Spec = direct.ValueOf(DataCatalogTagTemplateSpec_FromProto(mapCtx, a.actual))
+	if mapCtx.Err() != nil {
+		return nil, mapCtx.Err()
+	}
+	obj.Spec.ProjectRef = &refsv1beta1.ProjectRef{External: a.id.Parent().ProjectID}
+	obj.Spec.Location = a.id.Parent().Location
+	uObj, err := runtime.DefaultUnstructuredConverter.ToUnstructured(obj)
+	if err != nil {
+		return nil, fmt.Errorf("converting to unstructured: %w", err)
+	}
+
+	u.SetName(a.id.ID())
+	u.SetNamespace(a.desired.GetNamespace()) // Preserve namespace
+	u.SetGroupVersionKind(krm.DataCatalogTagTemplateGVK)
+
+	// Clear output-only fields from spec before setting object
+	// 'dataplex_transfer_status' is marked Optional but behaves like output-only in practice for export.
+	unstructured.RemoveNestedField(uObj, "spec", "dataplexTransferStatus")
```

Comment:
Not sure why this field is removed.

Reply:
ah this is in observed state. dont need this line.